pkg/notes/document/document.go

```diff
@@ -40,6 +40,52 @@ type Document struct {
 	PreviousRevision        string
 }
 
+// REVIEWER: This really should be just `Notes` but I want to get some feedback
+// on this idea before doing more refactoring along those lines. This container
+// is useful for several reasons:
+// (1) Its a single container to pass to the template which keesp the template simple.
+// (2) We can sort it easily before rendering.
+// (3) Because we use NoteCategory and a collection rendering in the template is simpler:
+//
+// {{range .NoteCollection}}
+// 	### {{.Kind | prettyKind}}
+//		{{- range $note := .Notes}} - $note {{-end}}
+// {{end}}
+//
+// Of course this assumes the note collection is sorted already which we'll do before rendering.
+
+// NoteCollection is a collection of note categories and implements sorted.Interface
+type NoteCollection []NoteCategory
+
+// Sort sorts selection by sorted by priority order.
+func (n *NoteCollection) Sort(kindPriority []Kind) error {
+	// MAYBE? Implement sorted.Interface instead?
```

Comment:
I assume you're referring the `sort.Interface`, right? What would be the benefit in implementing this interface?

Reply:
Yep. I left it as a place holder but I don't see a strong benefit to implementing sort.Interface to be honest. It was just a thought :)

Reply:
I'm happy with the `Sort` method, it behaves in the same way like the other functions inside the `sort` package :+1: